blocks/bricks/sequence_generators.py

```diff
@@ -136,6 +137,34 @@ def _push_allocation_config(self):
 
     @application
     def cost(self, application_call, outputs, mask=None, **kwargs):
+        """Returns sum of generation costs for output sequences.
+
+        Parameters
+        ----------
+        outputs : :class:`~tensor.TensorVariable`
+            The 3(2) dimensional tensor containing output sequences.
+            The dimension 0 must stand for time, the dimension 1 for the
+            position on the batch.
+        mask : :class:`~tensor.TensorVariable`
+            The binary matrix identifying fake outputs.
+
+        Notes
+        -----
+        The contexts are expected as keyword arguments.
+
+        """
+        # Compute the sum of costs
+        cost = self.cost_matrix(outputs, mask=mask, **kwargs).sum()
+        add_role(cost, COST)
+
+        # Add auxiliary variable for per sequence element cost
+        application_call.add_auxiliary_variable(
+            (cost / mask.sum()) if mask is not None else cost,
+            name='per_sequence_element')
```

Comment:
@orhanf Might be relying on the fact that the parameter name in the end will be `brick_application_name`, so it will probably be `sequence_generator_cost_per_sequence_element` in the end.